packages/app/src/elements/import-dialog.ts

```diff
@@ -40,14 +58,41 @@ export class ImportDialog extends Dialog<File, void> {
                 ></pl-select>
 
                 <div class="small padded" ?hidden=${this._formatSelect && this._formatSelect.value !== imp.CSV.value}>
-                    ${$l(
-                        "IMPORTANT: Before importing, please make sure that your CSV data " +
-                            "is structured according to {0}'s specific requirements!",
-                        process.env.PL_APP_NAME!
-                    )}
+                    ${$l("Choose the field type for each column. If you are having trouble,")}
                     <a href="#" @click=${this._downloadCSVSampleFile}> ${$l("Download Sample File")} </a>
                 </div>
 
+                <div
+                    class="vertical stretching spacing layout"
+                    ?hidden=${this._formatSelect && this._formatSelect.value !== imp.CSV.value}
+                >
+                    <ul>
+                        ${this._itemColumns.map(
+                            (itemColumn, itemColumnIndex) => html`
+                                <li class="margined">
+                                    <pl-select
+                                        id=${`itemColumnSelect-${itemColumnIndex}`}
+                                        icon=${FIELD_DEFS[itemColumn.type].icon}
+                                        .label=${itemColumn.displayName}
+                                        .options=${fieldTypeOptions}
+                                        .value=${itemColumn.type}
+                                        @change=${() => {
+                                            const thisElement = this.shadowRoot?.querySelector(
+                                                `#itemColumnSelect-${itemColumnIndex}`
+                                            ) as HTMLSelectElement;
```

Comment:
I know you're still sketching and probably aware of this, but this can be done more elegantly with the `@query` decorator (see the `_formatSelect` and `_vaultSelect` properties).

Reply:
Oh, nevermind... I just realised you have a select for each column so that won't work. Another option would be to use `@queryAll` and then `this._fieldTypeSelects[itemColumnIndex]`.

Reply:
(One of the advantages of using `@query` and `@queryAll` is that you can define the types of the element and in this case even of the value, so you wouldn't have to do any casting here. E.g.:

```ts
@queryAll("pl-select.field-type-select")
_fieldTypeSelects: Select<FieldType>[];
```
and then yo could simply do

```ts
const newValue = this._fieldTypeSelects[itemColumnIndex]?.value; // Already has type `FieldType`
```

Reply:
Thanks, I couldn't get `@query` to work, and didn't realize `@queryAll` could also be helpful. I'll try that as cleanup.